__tests__/Map.ts

```diff
@@ -97,6 +100,7 @@ describe('Map', () => {
   it('accepts non-collection array-like objects as keyed collections', () => {
     const m = Map({ length: 3, 1: 'one' });
     expect(m.get('length')).toBe(3);
+    // @ts-expect-error -- type error, but the API is tolerante
```

Comment:
This should not be a type error. Similarly, we should ensure that `m.get(1)` is well typed to return NSV (or be a type error if that arg is omitted)

I filed an issue about this a while back and it was closed as "Working as intended" despite being clearly incorrect. 🤷  - https://github.com/microsoft/TypeScript/issues/45170#issuecomment-909751159

A workaround type to fix this looks like:

```ts
type ObjectKeys<T> = { [K in keyof T]: K extends number ? `${K}` : K }[keyof T]
```